docs/source/python/pandas.rst

```diff
@@ -293,3 +293,19 @@ Used together, the call
 
 will yield significantly lower memory usage in some scenarios. Without these
 options, ``to_pandas`` will always double memory.
+
+Note that ``self_destruct=True`` is not guaranteed to save memory. Since the
+conversion happens column by column, memory is also freed column by column. But
+if multiple columns share an underlying allocation, then no memory will be
```

Comment:
What does "share" mean here? Do you mean different arrays share the same underlying buffer? How does that happen?

Reply:
Yes, multiple arrays may share the same buffer. In IPC for instance, we read a record batch's worth of data from the file at a time, and hence all arrays in that batch share the same buffer. In Flight, similarly, we receive a record batch's worth of data from gRPC and (for implementation reasons) concatenate it into a single buffer, so we end up in the same situation. (I don't think this applies to, say, Parquet or CSV, since there's actual decoding for those formats, but haven't tested it.)

Reply:
Ah, you're right, thanks.